pkg/services/baremetal/host/host.go

```diff
@@ -617,17 +618,16 @@ func (s *Service) actionRegistering() actionResult {
 }
 
 func validateRootDeviceWwnsAreSubsetOfExistingWwns(rootDeviceHints *infrav1.RootDeviceHints, storageDevices []infrav1.Storage) error {
+	knownWWNs := make([]string, 0, len(storageDevices))
```

Comment:
if you do this already, you should use a map and save some computing time compared to looping over slices in slices.Contains()

Reply:
@janiskemper 

> if you do this already, you should use a map and save some computing time compared to looping over slices in slices.Contains()

Are you sure a map is faster? A slice is sequential memory access, which is very fast. 

Here this gets explained: https://stackoverflow.com/a/31707228/633961

Reply:
not faster apparently, as you tried it, just fewer steps to do. But it is not very significant anyway

Reply:
@janiskemper what is missing for this PR to get your approval?